components/brave_wallet/browser/keyring_service.cc

```diff
@@ -834,47 +827,16 @@ void KeyringService::RestoreWallet(const std::string& mnemonic,
 
   if (IsFilecoinEnabled()) {
     // Restore mainnet filecoin acc
-    auto* filecoin_keyring =
-        RestoreKeyring(mojom::kFilecoinKeyringId, mnemonic, password, false);
-    if (filecoin_keyring && !GetDerivedAccountsNumberForKeyring(
-                                profile_prefs_, mojom::kFilecoinKeyringId)) {
-      auto address =
-          AddAccountForKeyring(mojom::kFilecoinKeyringId, GetAccountName(1));
```

Comment:
Is this intentional? If we don't create first Filecoin account and account discovery found nothing, we will end up have no Filecoin accounts created.

Reply:
Yes, previous behaviour was wrong